lms/djangoapps/student_enrollment/enrollment.py

```diff
@@ -103,12 +116,17 @@ def enroll(self):
                 if sample_content is not None:
                     sample_content.enroll_student_in_program(user.email)
                 if learning_supports:
-                    student_source = student["Student_Source"].strip(" \"\'")
+                    student_source = student["Student_Source"].strip(" \"\'") if student.get("Student_Source") else None
                     for prog in learning_supports:
-                        eligible_sources = list(map(lambda x: x.strip(" \'\"\r\n"),
-                                                    prog.support_program_sources.split(",")))
-                        if student_source in eligible_sources:
+                        # if eligible sources list is empty, treat all students as eligible and enrol
+                        if not prog.support_program_sources:
                             prog.enroll_student_in_program(user.email)
+                        # otherwise check student source against eligible sources before enrolling
+                        else:
+                            eligible_sources = list(map(lambda x: x.strip(" \'\"\r\n"),
+                                                    prog.support_program_sources.split(",")))
+                            if student_source in eligible_sources:
```

Comment:
Are support programs all program ids? Is there a reason we're not, adding ForeignKeys or something instead?

Reply:
Yes they are all programme ids (program codes). The reason is a) a very short deadline for this (I essentially learned about this requirement 2 days before the Christmas break, and it either needs to go live by the end of this week, or not at all) and b) there is going to be more than one support programme for l3bootcamp, which means it would need to be a ManyToMany, not a ForeignKey - a refactor which I did not feel comfortable implementing in such a short timeframe.